ocb3/src/lib.rs

```diff
@@ -0,0 +1,602 @@
+#![no_std]
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![doc = include_str!("../README.md")]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo.svg",
+    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo.svg"
+)]
+#![deny(unsafe_code)]
+#![warn(missing_docs, rust_2018_idioms)]
+#![allow(non_snake_case)]
+
+use core::marker::PhantomData;
+
+pub use aead::{
+    self, generic_array::GenericArray, AeadCore, AeadInPlace, Error, KeyInit, KeySizeUser,
+};
+use aes::{self, Aes128, Aes256, Block};
+use cipher::{
+    consts::{U0, U12, U16},
+    BlockDecrypt, BlockEncrypt, BlockSizeUser,
+};
+
+mod util;
+use crate::util::{double, inplace_xor, ntz};
+use subtle::ConstantTimeEq;
+
+/// Number of L values to be precomputed. Precomputing m values, allows
+/// processing inputs of length up to 2^m blocks (2^m * 16 bytes) without
+/// needing to calculate L values at runtime.
+///
+/// By setting this to 32, we can process inputs of length up to 1 terabyte.
+const L_TABLE_SIZE: usize = 32;
+
+/// Max associated data.
+pub const A_MAX: usize = 1 << (L_TABLE_SIZE + 4);
+/// Max plaintext.
+pub const P_MAX: usize = 1 << (L_TABLE_SIZE + 4);
+/// Max ciphertext.
+pub const C_MAX: usize = 1 << (L_TABLE_SIZE + 4);
+
+/// OCB3 nonce
+pub type Nonce<NonceSize> = GenericArray<u8, NonceSize>;
+
+/// OCB3 tag
+pub type Tag<TagSize> = GenericArray<u8, TagSize>;
+
+/// AES-OCB3 with a 128-bit key, 96-bit nonce, and 128-bit tag.
+pub type Aes128Ocb3 = AesOcb3<Aes128, U12>;
```

Comment:
You don't feature gate these aliases properly. Personally, I would prefer for this crate to be cipher-agnostic, maybe later we could introduce `aes-ocb3` as a helper combination crate.